crates/revm/src/optimism/mod.rs

```diff
@@ -94,6 +97,65 @@ pub fn parse_l1_info_tx(data: &[u8]) -> Result<L1BlockInfo, BlockExecutionError>
     Ok(l1block)
 }
 
+/// Parses the calldata of the [L1BlockInfo] transaction post-Ecotone hardfork.
+pub fn parse_l1_info_tx_ecotone(data: &[u8]) -> Result<L1BlockInfo, BlockExecutionError> {
+    // The setL1BlockValuesEcotone tx calldata must be exactly 260 bytes long, considering that
+    // we already removed the first 4 bytes (the function selector). Detailed breakdown:
+    //   8 bytes for the block sequence number
+    // + 4 bytes for the blob base fee scalar
+    // + 4 bytes for the base fee scalar
+    // + 8 bytes for the block number
+    // + 8 bytes for the block timestamp
+    // + 32 bytes for the base fee
+    // + 32 bytes for the blob base fee
+    // + 32 bytes for the block hash
+    // + 32 bytes for the batcher hash
+    if data.len() != 160 {
+        return Err(reth_executor::BlockExecutionError::OptimismBlockExecution(
+            reth_executor::OptimismBlockExecutionError::L1BlockInfoError {
+                message: "unexpected l1 block info tx calldata length found".to_string(),
+            },
+        ))
+    }
+
+    let l1_base_fee = U256::try_from_be_slice(&data[32..64]).ok_or(
```

Comment:
any reason these are not ordered in the same manner as the comment above?

Reply:
No reason; re-ordered :+1: